client/websocket_client.tsx

```diff
@@ -176,28 +189,33 @@ export default class WebSocketClient {
         }
     }
 
-    userTyping(channelId, parentId, callback) {
-        const data = {};
+    userTyping(channelId: string, parentId: string, callback?: () => void) {
+        const data = {} as {
+            channel_id: string,
+            parent_id: string,
+        }; // TODO is this a defined type?
         data.channel_id = channelId;
         data.parent_id = parentId;
```

Comment:
This can be done as one statement to simplify typing.
```suggestion
        const data = {
            channel_id: channelId,
            parent_id: parentId,
        };
```